### PR TITLE
Added all Minecraft Tags

### PR DESCRIPTION
**Pull request by TheBusyBiscuit**
# Description
Sorry about the 170 changed files :eyes:

This Pull Request adds all Minecraft Tags for version 1.16.3 to MockBukkit.
I added a TagParser and a Tag Wrapper which can read the tag format according to the json standard described here:
https://minecraft.gamepedia.com/Tag#JSON_format

I have also added various unit tests, including a check for cyclic / infinitely looped tags.
And a safety check to ensure that the Tags default behaviour hasn't been disturbed.
This removes the need to register tags manually and allows a developer to just access the tags normally as they would with Spigot/Paper.

I also restored support for Tag registries by creating an enum for it and handling tags seperately based on their registry.

Also, I bumped the version to 0.7.0 already since I am quite confident that one of the other two pull requests will get merged first XD

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Version number updated in `settings.gradle`.
- [x] Unit tests added.
- [x] Code follows existing code format.

# Info on creating a pull request
Before the pull request can be accepted, the version number stored in settings.gradle should be updated.
MockBukkit uses semantic-versioning (https://semver.org/)

 - If the PR fixes a bug in MockBukkit, update the patch version. (if the current version is `0.4.1`, the new version should be `0.4.2`)
 - If the PR adds a new feature to MockBukkit, update minor version. (if the current version is `0.4.1`, the new version should be `0.5.0`)
 - Make sure that unit tests are added which test the relevant changes.
 - Make sure that the changes follow the existing code format.

Note that a PR that fixes an `UnimplementedOperationException` should be considered a new version and not a bugfix.

The version number can be found in `settings.gradle`.

### CODE REVIEW

File: src/main/java/be/seeseemelk/mockbukkit/entity/LivingEntityMock.java [hunk 1]
```diff
@@ -629,6 +629,5 @@ public void setInvisible(boolean invisible)
 	public boolean isInvisible()
 	{
 		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
-	}
+		throw new UnimplementedOperationException()
```
Comment (poma123) [inline src/main/java/be/seeseemelk/mockbukkit/entity/LivingEntityMock.java]: The build is failing because of this change. Is this intended?

**Review by poma123 — CHANGES_REQUESTED** [review]

File: src/main/java/be/seeseemelk/mockbukkit/entity/LivingEntityMock.java (hunk 1 above)
Comment (TheBusyBiscuit) [reply src/main/java/be/seeseemelk/mockbukkit/entity/LivingEntityMock.java]: Missing semicolon XD
Thanks for noticing @poma123, I simply resolved the conflicts on mobile and didn't even bother to check the CI results.

**Comment by seeseemelk** [comment]
> Also, I bumped the version to 0.7.0 already since I am quite confident that one of the other two pull requests will get merged first XD

Well, you can bump it up to `0.8.0` now :smile:

**Comment by TheBusyBiscuit** [comment]
> > Also, I bumped the version to 0.7.0 already since I am quite confident that one of the other two pull requests will get merged first XD
> 
> Well, you can bump it up to `0.8.0` now 😄

Pfff, I'm way ahead of ya 😎

**Review by seeseemelk — APPROVED** [review]

**Pull request merged**